### PR TITLE
cpu/stm32: stm32f4 BRR from BSRR

### PR DESCRIPTION
**Pull request by kfessel**
### Contribution description

sometimes one wants to save one instruction :) 
just write the bits we need to write.

### Testing procedure

tests/periph/gpio_ll tests this 

### Issues/PRs references

@maribu might know some reference

maybe #19407

### CODE REVIEW

File: cpu/stm32/include/gpio_ll_arch.h [hunk 1]
```diff
@@ -77,7 +77,8 @@ static inline void gpio_ll_clear(gpio_port_t port, uword_t mask)
 #if defined(GPIO_BRR_BR0) && !defined(CPU_FAM_STM32F4)
     p->BRR = mask;
 #else
-    p->BSRR = mask << 16;
+    uint16_t * brr = (void*)&(p->BSRR);
```
Comment (maribu) [inline cpu/stm32/include/gpio_ll_arch.h:80]:
```suggestion
    uint16_t *brr = (void *)&(p->BSRR);
```

File: cpu/stm32/include/gpio_ll_arch.h [hunk 2]
```diff
@@ -77,7 +77,8 @@ static inline void gpio_ll_clear(gpio_port_t port, uword_t mask)
...
+    brr[1] = (uint16_t) mask;
```
Comment (maribu) [inline cpu/stm32/include/gpio_ll_arch.h:81]:
```suggestion
    brr[1] = (uint16_t)mask;
```

**Review by maribu — COMMENTED** [review]
That should indeed save an instruction. Previously the compiler would take the GPIO port base address

1. add the offset of the BSRR register 2. do a left shift
3. do a 32 bit store

Now it would take the GPIO port base address
1. add the offest of the BSRR register + 2
2. do a 16 bit store

Please fix the style nitpicks and squash directly

**Review by maribu — APPROVED** [review]

**Comment by benpicco** [comment]
bors merge

File: cpu/stm32/include/gpio_ll_arch.h [hunk 3]
```diff
@@ -77,7 +77,8 @@ static inline void gpio_ll_clear(gpio_port_t port, uword_t mask)
 #if defined(GPIO_BRR_BR0) && !defined(CPU_FAM_STM32F4)
     p->BRR = mask;
 #else
-    p->BSRR = mask << 16;
+    uint16_t *brr = (void *)&(p->BSRR);
```
Comment (maribu) [inline cpu/stm32/include/gpio_ll_arch.h:80]:
```suggestion
    volatile uint16_t *brr = (volatile uint16_t *)&(p->BSRR);
```

The `volatile` is needed here abd C++ is being annoying again here.

**Comment by maribu** [comment]
bors cancel

**Pull request merged**